### PR TITLE
Fixed logging messages

### PR DESCRIPTION
**Pull request by Conorbro**
Fix format of logging messages.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -44,27 +42,27 @@ func doScrape(request *http.Request, client *http.Client, logger log.Logger) {
 	scrapeResp, err := client.Do(request)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to scrape %s: %s", request.URL.String(), err)
-		level.Warn(logger).Log(msg)
+		level.Warn(logger).Log("msg", msg, "scrap_id", request.Header.Get("id"))
```
Comment (brian-brazil) [inline client/client.go]: The `msg` variable here is not structured logging, so you'll have to construct it by hand.

scrape_id

File: client/client.go [hunk 2]
```diff
@@ -44,27 +42,27 @@ func doScrape(request *http.Request, client *http.Client, logger log.Logger) {
...
-	level.Info(logger).Log("Retrieved scrape response")
+	level.Info(logger).Log("msg", "Retrieved scrape response", "scrap_id", request.Header.Get("id"))
 	err = doPush(scrapeResp, request, client)
 	if err != nil {
-		level.Warn(logger).Log("Failed to push scrape response:", err)
+		level.Warn(logger).Log("msg", "Failed to push scrape response:", err, "scrap_id", request.Header.Get("id"))
 		return
 	}
-	level.Info(logger).Log("Pushed scrape result")
+	level.Info(logger).Log("msg", "Pushed scrape result", "scrap_id", request.Header.Get("id"))
```
Comment (brian-brazil) [inline client/client.go]: Given all these logs have the same scrape_id field, it'd be simpler to use With on the logger to add it

File: client/client.go [hunk 3]
```diff
@@ -96,13 +94,13 @@ func loop(logger log.Logger) {
 	client := &http.Client{}
 	resp, err := client.Post(*proxyURL+"/poll", "", strings.NewReader(*myFqdn))
 	if err != nil {
-		level.Info(logger).Log("Error polling:", err)
+		level.Error(logger).Log("msg", "Error polling:", err)
```
Comment (brian-brazil) [inline client/client.go]: There should always be an even number of fields. Here you want to add `"err"`.

File: client/client.go [hunk 4]
```diff
@@ -116,10 +114,9 @@ func main() {
 	kingpin.Parse()
 	logger := promlog.New(allowedLevel)
 	if *proxyURL == "" {
-		level.Error(logger).Log("-proxy-url flag must be specified.")
+		level.Error(logger).Log("msg", "-proxy-url flag must be specified.")
```
Comment (brian-brazil) [inline client/client.go]: This should exit the program.

File: proxy/proxy.go [hunk 5]
```diff
@@ -55,7 +55,7 @@ func main() {
 
 			resp, err := coordinator.DoScrape(ctx, request)
 			if err != nil {
-				level.Error(logger).Log("url", "Error scraping: %s", err)
+				level.Error(logger).Log("msg", "Error scraping:", err, "url", request.URL.String())
```
Comment (brian-brazil) [inline proxy/proxy.go]: err again

**Comment by Conorbro** [comment]
@brian-brazil

**Pull request merged**